core/blockchain_test.go

```diff
@@ -706,14 +729,16 @@ func TestTransactionIndices(t *testing.T) {
 	chain.DrainAcceptorQueue()
 
 	chain.Stop()
-	check(nil, chain) // check all indices has been indexed
+	check(nil, false, chain.CurrentBlock().Number.Uint64(), chain) // check all indices has been indexed
 
 	lastAcceptedHash := chain.CurrentHeader().Hash()
 
 	// Reconstruct a block chain which only reserves limited tx indices
 	// 128 blocks were previously indexed. Now we add a new block at each test step.
-	limit := []uint64{130 /* 129 + 1 reserve all */, 64 /* drop stale */, 32 /* shorten history */}
-	tails := []uint64{0 /* reserve all */, 67 /* 130 - 64 + 1 */, 100 /* 131 - 32 + 1 */}
+	limit := []int64{130 /* 129 + 1 reserve all */, 64 /* drop stale */, 32 /* shorten history */, -1 /* skip indexing */, -1 /* skip indexing */}
+	tails := []uint64{0 /* reserve all */, 67 /* 130 - 64 + 1 */, 100 /* 131 - 32 + 1 */, 100 /* skip indexing */, 100 /* skip indexing */}
+	// lastIndexed should stop with limit=-1 otherwise it should be the current block number
+	lastIndexed := []uint64{129 /* current block number */, 130 /* current block number */, 131 /* current block number */, 131 /* skip indexing */, 131 /* skip indexing */}
```

Comment:
we should convert this to a struct in the future. It was a bit hard to follow before and going from 2 to 3 values makes it a bit harder to follow.

Reply:
might be nice to do this as part of this PR (or immediate follow-up), doesn't seem like a big change.